iree/hal/cts/driver_test.cc

```diff
@@ -30,7 +30,7 @@ TEST_P(DriverTest, CreateDefaultDevice) {
 TEST_P(DriverTest, EnumerateAndCreateAvailableDevices) {
   IREE_ASSERT_OK_AND_ASSIGN(auto devices, driver_->EnumerateAvailableDevices());
 
-  for (int i = 0; i < devices.size(); ++i) {
+  for (size_t i = 0; i < devices.size(); ++i) {
```

Comment:
We've got a bunch of these type mismatches today 😅 
Running under different compilers can improve code quality by shining a light on them, so this is nice!
nit: such fixes could be moved out to their own PR, to keep this one focused on the Metal/BUILD specific changes (which will be more contentious)

Reply:
Got it. Will remove them.